airflow/contrib/hooks/gcs_hook.py

```diff
@@ -175,40 +175,55 @@ def download(self, bucket_name, object_name, filename=None):
 
         return blob.download_as_string()
 
-    def upload(self, bucket_name, object_name, filename,
-               mime_type='application/octet-stream', gzip=False):
+    def upload(self, bucket_name, object_name, filename=None,
+               data=None, mime_type=None, gzip=False):
         """
-        Uploads a local file to Google Cloud Storage.
-
+        Uploads a local file or file content as a string or bytes to Google Cloud Storage.
         :param bucket_name: The bucket to upload to.
         :type bucket_name: str
         :param object_name: The object name to set when uploading the local file.
         :type object_name: str
         :param filename: The local file path to the file to be uploaded.
         :type filename: str
+        :param data: The file's content as a string or bytes to be uploaded.
+        :type data: str
         :param mime_type: The MIME type to set when uploading the file.
         :type mime_type: str
-        :param gzip: Option to compress file for upload
+        :param gzip: Option to compress local file for upload
```

Comment:
Why didn't you add compression support for data?
```
            if gzip:
                out = StringIO.StringIO()
                with gz.GzipFile(fileobj=out, mode="w") as f:
                    f.write(data)
                data = out.getvalue()
```

Reply:
I actually had a use case at work where compression couldn’t be used when uploading files to BigQuery from a Storage Bucket. I didn’t think to add it but that is a great idea. I will add it asap. Thank you